engines/grim/pool.h

```diff
@@ -314,8 +314,9 @@ void PoolObject<T>::Pool::restoreObjects(SaveGame *state) {
 	Common::HashMap<int32, T *> tempMap;
 	for (int32 i = 0; i < size; ++i) {
 		int32 id = state->readLESint32();
-		T *t = _map.getVal(id);
-		_map.erase(id);
+		T *t = nullptr;
+		if (_map.tryGetVal(id, t))
+			_map.erase(id);
 		if (!t) {
```

Comment:
I would've probably reused the existing if statement. It's clear from context that it's the same condition.

Reply:
I didn't know if I could assume that `_map` does not contain null pointers. But if that is a safe assumption we could indeed use an `else` statement here.

Reply:
It calls a member function only a few lines bellow, no null pointers could've been stored legitimately.

Reply:
not sure about save game and I'm not sure about this logic here. I didn't add this pool template.
I agree with henke37 assumption that map does not have null pointers. Also I agree with use 'else'.

Reply:
I think the map indeed never contains null pointers. Thus I have changed the code to use `else`, which is a bit clearer.